File: searching/javascript/linear.js
/*
*Time Complexity= O(n)
*starts from one end and search till the last element untill the desired element is found
*/

module.exports = {
    linearSearch : linearSearch
}

function linearSearch(data,l,n,value){
    for(var i=0;i<1000000;i++){
        if(data[i]==value){
            return "find first at index "+i;
        }
    }
    return "element does not exists";   
}


//standalone

if(typeof require && require.main == module){
    var data=[];
    var value;
    var i=0;
    var n;
    
    console.log("enter number of elements:");
    var stdin = process.openStdin();
    stdin.addListener("data", function(answer) {
        answer = parseInt(answer.toString().trim());
        if(n === undefined){
            n = answer;
            console.log("enter elements:");
        }else if(i < n-1){
            //console.log("i "+i);
            data[i] = answer;
            i++;
        }else if(n-1 == i){
            data[i] = answer;
            i++;
            console.log("enter key value to search:");
        }else{
            value = answer;
            console.log(data);
            console.log(value);
            result = linearSearch(data,0,n,value);
            console.log(result);
            n=undefined;
            console.log("enter the number of elements:");
            i=0;
            data=[];
        }
    });
}

File: searching/java/linear.java
import java.util.Scanner;
public class Linear {
    public static void main(String[] args) {
        int a[]=new int[1000000];
        System.out.print("Enter the number of Elements:: ");
        Scanner d=new Scanner(System.in);
        int n=d.nextInt();
        System.out.print("Enter the arrey:: ");
        for(int i=0;i<n;i++){
            a[i]=d.nextInt();
        }
        System.out.print("Enter the element do you went to search::  ");
        int p=d.nextInt();
        int z=Linear.linearSearch(a,p);
        if(z!=0)
        System.out.println("Element fount at "+ (z+1) + " position.");
        else{
            System.out.print("Not Found.");
        }
    }
   public static int linearSearch(int a[],int search){
        for(int i=0;i<a.length;i++){
            if(a[i]==search){
                return i;
            }
        }
        return 0;
    }

}
    

